perceval/backend.py

```diff
@@ -510,6 +533,206 @@ def setup_cmd_parser(cls):
+        this archive will be removed to avoid corrupted archives.
+
+        The parameters needed to get the items are given using the
+        `backend_args` dict parameter.
+
+        :param backend_args: dict of arguments needed to fetch the items
+        :param category: category of the items to retrieve.
+           If None, it will use the default backend category
+        :param filter_classified: remove classified fields from the resulting items
+        :param manager: archive manager needed to store the items
+
+        :returns: a generator of items
+        """
+        if category:
+            backend_args['category'] = category
+        if filter_classified:
+            backend_args['filter_classified'] = filter_classified
+
+        fetch_args = find_signature_parameters(self.backend.fetch,
+                                               backend_args)
+        items = self.backend.fetch(**fetch_args)
+
+        try:
+            for item in items:
+                yield item
+        except Exception as e:
+            if manager:
+                archive_path = self.backend.archive.archive_path
+                manager.remove_archive(archive_path)
+            raise e
+
+    def __fetch_from_archive(self, manager, category, archived_after):
+        """Fetch items from an archive manager.
+
+        Generator to get the items of a category (previously fetched
+        by the given backend class) from an archive manager. Only those
+        items archived after the given date will be returned.
+
+        :param manager: archive manager where the items will be retrieved
+        :param category: category of the items to retrieve
+        :param archived_after: return items archived after this date
+
+        :returns: a generator of archived items
+        """
+        filepaths = manager.search(self.backend.origin,
+                                   self.backend.__class__.__name__,
+                                   category,
+                                   archived_after)
+
+        for filepath in filepaths:
+            self.backend.archive = Archive(filepath)
+            items = self.backend.fetch_from_archive()
+
+            try:
+                for item in items:
+                    yield item
+            except ArchiveError as e:
+                logger.warning("Ignoring %s archive due to: %s", filepath, str(e))
+
+
+class Summary:
+    """Summary class for fetch executions.
+
+    This class models the summary of a fetch execution. It includes
+    the last UUID, number of items fetched, skipped and their sum,
+    plus the min, max and last updated_on times. Furthermore, for
+    backends using offsets, the corresponding summary contains the
+    min and max offsets retrieved. Finally, the summary also includes
+    some extra fields, which can be used by any backend to include
+    fetch-specific information.
+    """
+    def __init__(self):
+        self.fetched = 0
```

Comment:
`fetched` should be renamed to something else. The backend might fetch a lot of items but for many reasons they wouldn't be generated.

Reply:
I don't see this change, `fetched` seems a reasonable definition.